Source/WebCore/rendering/svg/RenderSVGResourceGradient.h

```diff
@@ -31,6 +31,33 @@ namespace WebCore {
 
 class GraphicsContext;
 
+struct GradientData {
+    WTF_MAKE_STRUCT_FAST_ALLOCATED;
+
+    struct Inputs {
+        bool operator==(const Inputs& other)
+        {
+            return objectBoundingBox == other.objectBoundingBox && textPaintingScale == other.textPaintingScale;
+        }
+
+        std::optional<FloatRect> objectBoundingBox;
+        float textPaintingScale = 1;
+    };
+
+    bool validate(const Inputs& inputs)
+    {
+        if (this->inputs != inputs) {
+            gradient = nullptr;
+            this->inputs = inputs;
+        }
+        return gradient;
```

Comment:
What is the meaning of this return value? I would expect it will return whether the object was changed or not. Assume this sequence of calls:

```
auto data = makeUnique<GradientData>();
auto inputs = computeInputs(renderer, resourceMode);
auto v1 = data.validate(inputs);
auto v2 = data.validate(inputs);
```

Both v1 and v2 will be false although the object data was changed with v1 and it did not change with v2.

Reply:
I was thinking that the validate function returns whether the object is valid. But maybe the inverse would be better: name this function "invalidate" (or "invalidateIfNeeded"?), and return whether it did invalidate it.